src/views/LimitOrders/hooks/useGelatoLimitOrdersHistory.ts

```diff
@@ -1,28 +1,57 @@
-import { Order } from '@gelatonetwork/limit-orders-lib'
+import { Order, GelatoLimitOrders } from '@gelatonetwork/limit-orders-lib'
 import useActiveWeb3React from 'hooks/useActiveWeb3React'
 import useSWR from 'swr'
 import { SLOW_INTERVAL } from 'config/constants'
 import { useMemo } from 'react'
 
-import { getLSOrders, saveOrder } from 'utils/localStorageOrders'
+import { getLSOrders, saveOrder, saveOrders, hashOrderSet, hashOrder } from 'utils/localStorageOrders'
 import useGelatoLimitOrdersLib from 'hooks/limitOrders/useGelatoLimitOrdersLib'
 
-import { ORDER_CATEGORY } from '../types'
+import { ORDER_CATEGORY, LimitOrderStatus } from '../types'
 
 function newOrdersFirst(a: Order, b: Order) {
   return Number(b.updatedAt) - Number(a.updatedAt)
 }
 
-function syncOrderToLocalStorage({ chainId, account, orders }) {
-  const ordersLS = getLSOrders(chainId, account)
-
-  orders.forEach((order: Order) => {
-    const orderExists = ordersLS.find((confOrder) => confOrder.id.toLowerCase() === order.id.toLowerCase())
+const isOrderUpdated = (oldOrder: Order, newOrder: Order): boolean => {
+  return newOrder ? Number(oldOrder.updatedAt) < Number(newOrder.updatedAt) : false
+}
 
-    if (!orderExists || (orderExists && Number(orderExists.updatedAt) < Number(order.updatedAt))) {
-      saveOrder(chainId, account, order)
+async function syncOrderToLocalStorage({
+  gelatoLimitOrders,
+  chainId,
+  account,
+  orders,
+  syncTypes,
+}: {
+  gelatoLimitOrders: GelatoLimitOrders
+  chainId: number
+  account: string
+  orders: Order[]
+  syncTypes: LimitOrderStatus[]
+}) {
+  const allOrdersLS = getLSOrders(chainId, account)
+
+  const lsOrdersHashSet = hashOrderSet(allOrdersLS)
+  const newOrders = orders.filter((order: Order) => !lsOrdersHashSet.has(hashOrder(order)))
+  saveOrders(chainId, account, newOrders)
+
+  const typeOrdersLS = allOrdersLS.filter((order) => syncTypes.some((type) => type === order.status))
+
+  for (let i = 0; i < typeOrdersLS.length; i++) {
```

Comment:
for History Order Tab, we don't have to take this update step, right ?

Reply:
Yes, fixed